apps/meteor/server/modules/notifications/notifications.module.ts

```diff
@@ -173,9 +181,9 @@ export class NotificationsModule {
 			if (!this.userId) {
 				return false;
 			}
+			const canAccess = await Authorization.canAccessRoom(room, { _id: this.userId }, extraData);
```

Comment:
since you have only `rid`, you can use `Authorization.canAccessRoomId` , so you don't  need to add the query added on line 164.